From 0018de516fbc0304257406c68427204925ad7e9e Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Fri, 25 Oct 2024 15:53:12 +0200
Subject: [PATCH 700/841] feat(stm32mp2): fix CPU1 wake-up from RUN2

Allow CPU1 to wake-up from RUN2.
Without the flag RCC_CPUBOOTCR_BOOT_CPU1 set,
the CPU1 is set in HOLD BOOT after wake-up from
D1-Standby with CPU2 running.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 02cb476c952c255b59baf900f78ec0a86ca66a89 feat(stm32mp2): fix CPU1 wake-up from RUN2]
Change-Id: I472b1a9941bc13ce36134c06c254201090536d59
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/414303
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index a4fd6ff71..7e6e45c02 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -1293,6 +1293,9 @@ static void stm32_pm_init(void *fdt)
 	/* RCC init: DDR is shared by default */
 	mmio_setbits_32(rcc_base + RCC_DDRITFCFGR, RCC_DDRITFCFGR_DDRSHR);
 
+	/* Allow CPU1 (A35) to wake-up from D1 DStop1 */
+	mmio_setbits_32(rcc_base + RCC_CPUBOOTCR, RCC_CPUBOOTCR_BOOT_CPU1);
+
 	/* Legacy mode: only CPU1 is allowed to boot, core1 is OFF */
 	mmio_setbits_32(rcc_base + RCC_LEGBOOTCR, RCC_LEGBOOTCR_LEGACY_BEN);
 
-- 
2.47.2

